tests/cli/test_lib.py

```diff
@@ -55,6 +56,19 @@ def ReadOutput(self):
 
     return output_data
 
+  def ReadOutputString(self):
```

Comment:
Why would you compare against string when what the original intent was to compare against the resulting encoded data.

Reply:
This class has a Write() method that takes a string, so it seems to me that this class deals with strings, not bytes. 

I don't think it's desirable to have output tests being dependent on the implementation of the writer class.

Test should need to care that the strings they expect are the ones they get, not how the strings are encoded when output. That sort of testing belongs in test for the writers themselves.

Reply:
in that case maybe we should implement a test string output writer instead of encoding and decoding back and forth?

Reply:
Opened #1963 to track that